src/StarkEx.Crypto.SDK/Hashing/SpotTradingMessageHasher.cs

```diff
@@ -88,21 +111,79 @@ public BigInteger EncodeTransferWithFees(EncodeTransferWithFeesModel encodeTrans
+        Guards.NotNullOrEmptyOrWhitespace(encodeTransferWithoutFeesModel.ReceiverStarkKey);
+        Guards.NotInvalidHex(encodeTransferWithoutFeesModel.ReceiverStarkKey, nameof(encodeTransferWithoutFeesModel.ReceiverStarkKey));
+
+        var packedMessage = BigInteger.One
+            .ShiftLeft(31).Add(encodeTransferWithoutFeesModel.VaultIdFromSender)
+            .ShiftLeft(31).Add(encodeTransferWithoutFeesModel.VaultIdFromReceiver)
+            .ShiftLeft(63).Add(encodeTransferWithoutFeesModel.QuantizedAmountToTransfer)
+            .ShiftLeft(63).Add(BigInteger.Zero)
+            .ShiftLeft(31).Add(new BigInteger(encodeTransferWithoutFeesModel.Nonce.ToString()))
+            .ShiftLeft(22).Add(new BigInteger((encodeTransferWithoutFeesModel.ExpirationTimestamp / 3600).ToString()));
+
+        var firstInnerHash = pedersenHash.CreateHash(
+            new BigInteger(encodeTransferWithoutFeesModel.AssetIdSold.RemoveHexPrefix(), 16),
+            new BigInteger(encodeTransferWithoutFeesModel.ReceiverStarkKey.RemoveHexPrefix(), 16));
+
+        return pedersenHash.CreateHash(firstInnerHash, packedMessage);
+    }
+
     /// <inheritdoc />
     public BigInteger EncodeConditionalTransferWithFees(EncodeTransferWithFeesModel encodeTransferWithFeesModel)
     {
+        Guards.NotNullOrEmptyOrWhitespace(encodeTransferWithFeesModel.AssetIdSold);
+        Guards.NotInvalidHex(encodeTransferWithFeesModel.AssetIdSold, nameof(encodeTransferWithFeesModel.AssetIdSold));
+        Guards.NotNullOrEmptyOrWhitespace(encodeTransferWithFeesModel.AssetIdUsedForFees);
+        Guards.NotInvalidHex(encodeTransferWithFeesModel.AssetIdUsedForFees, nameof(encodeTransferWithFeesModel.AssetIdUsedForFees));
+        Guards.NotNullOrEmptyOrWhitespace(encodeTransferWithFeesModel.ReceiverStarkKey);
+        Guards.NotInvalidHex(encodeTransferWithFeesModel.ReceiverStarkKey, nameof(encodeTransferWithFeesModel.ReceiverStarkKey));
         Guards.NotNull(encodeTransferWithFeesModel.Fact);
         Guards.NotNull(encodeTransferWithFeesModel.FactRegistryAddress);
         Guards.NotInvalidHex(encodeTransferWithFeesModel.Fact, nameof(encodeTransferWithFeesModel.Fact));
         Guards.NotInvalidHex(encodeTransferWithFeesModel.FactRegistryAddress, nameof(encodeTransferWithFeesModel.FactRegistryAddress));
 
-        var condition = pedersenHash.CreateHash(
-            new BigInteger(encodeTransferWithFeesModel.Fact.RemoveHexPrefix(), 16),
-            new BigInteger(encodeTransferWithFeesModel.FactRegistryAddress.RemoveHexPrefix(), 16));
+        var condition = GetCondition(encodeTransferWithFeesModel.Fact, encodeTransferWithFeesModel.FactRegistryAddress);
 
         return EncodeConditionalTransferWithFees(encodeTransferWithFeesModel, condition.ToByteArray().ToHex());
     }
 
+    /// <inheritdoc />
+    public BigInteger EncodeConditionalTransferWithoutFees(EncodeTransferWithoutFeesModel encodeTransferWithoutFeesModel)
```

Comment:
This is functionally equivalent to `DeprecatedHashConditionalTransfer`.

Reply:
Deleted the existing deprecated method in favor of this one